PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+            debug += "❌ " + "Error " + JSON.stringify(content["error"]) + " | " 
+            fields_for_debug.push(debug)
+            DEBUG_SHEET.appendRow(fields_for_debug);
+        }
+        else{
+            // Build a dic that will contain all the extracted information
+            let output = {
+              "Date": new Date().toISOString().slice(0, 10),
+              "Label": batch[i].label,
+              "URL": batch[i].url,
+              "Device": batch[i].device,
+              "URL / Origin": batch[i].origin
+            };
+            // Iterate through all the fields and eval the function to get their respective values, adding them to the dic
+            for (let j = 0; j < fields.length; j++) {
+                try{
+                    // Origin field is used in some calls to know if there was no URL-level data and we reverted to Origin level
+                    let origin = batch[i].origin;
+                    let green_domains_ = green_domains;
+                    // Eval the formula, for example "content["lighthouseResult"]["lighthouseVersion"]"
+                    let value = eval(fields[j].data)
+                    // Add it to our dic
+                    output[fields[j].field] = value
+                }
+                catch(e){
+                    // If error, write it in placce -> we ultimately decided to leave it blank instead as it was breaking dashboarding down the line
+                    // output[fields[j].field] = "Error " + JSON.stringify(e.toString())
+                    output[fields[j].field] = ""
+                    debug += "❗️ Error on " + fields[j].field + JSON.stringify(e.toString()) + " | " 
+                }
+            }
+            Logger.log(output);
+            
+            // STEP 3. We build the new row based on what we read on the Results first line
+            if(api == "PSI"){
+              let sheet_results = RESULT_SHEET;
+              let results_header = RESULT_SHEET_HEADER;
+              // We translate the dic output to an array row based on indexes found in "Results"
+              let array = [];
+              // Go through all the Dimensions listed in the header and build the array
+              for (let k = 0; k < results_header[0].length; k++){
+                let dimension = results_header[0][k];
+                if(dimension in output){
+                  array.push(output[dimension])
+                }
+                else{
+                  // We ultimately decided to leave blank columns without data and leave the details in the debugging.
+                  // array.push("No data")
+                  array.push("")
+                  debug += "💢 Error on " + dimension.toString() + " - No data" + " | " 
+                }
+              }
+              // Append the array at the end of "Results"
+              sheet_results.appendRow(array)
+            }
+            else if(api == "CrUX"){
```

Comment:
This is the same as above when `api === "PSI"`

Reply:
done!

Reply:
I meant you could use an or in the PSI case instead of repeating the code. The debugging code shouldn't be checked in anyhow.

This is likely the case for everything except CrUX History. Maybe use an if to set `sheet_results` and `results_header` for the green domain, accessibility, sustainability cases instead of repeating the code.